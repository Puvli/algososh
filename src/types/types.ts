import { ElementStates } from "./element-states";

export type TDataElement = {
  value: string | number;
  state: ElementStates;
  pos?: boolean;
  valueToChange?: string | number;
  isHead?: boolean;
  isTail?: boolean;
  isLinked?: boolean;
};

export enum SortAlgorithm {
  selectsort = "выбором",
  bubble = "пузырьком",
}
