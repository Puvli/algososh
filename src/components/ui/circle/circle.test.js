import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { Button } from "../button/button";
import { ElementStates } from "../../../types/element-states";

const TEST_SNAPSHOT_CIRCLE_WITHOUT_LETTER = "Рендер круга без буквы";
const TEST_SNAPSHOT_CIRCLE_WITH_LETTERS = "Рендер круга c буквами";
const TEST_SNAPSHOT_CIRCLE_WITH_HEAD = "Рендер круга c head";
const TEST_SNAPSHOT_CIRCLE_WITH_HEAD_REACT = "Рендер круга c head react";
const TEST_SNAPSHOT_CIRCLE_WITH_TAIL = "Рендер круга c tail";
const TEST_SNAPSHOT_CIRCLE_WITH_TAIL_REACT = "Рендер круга c tail react";
const TEST_SNAPSHOT_CIRCLE_WITH_INDEX = "Рендер круга c index";
const TEST_SNAPSHOT_CIRCLE_WITH_SMALL = "Рендер круга c small";
const TEST_SNAPSHOT_CIRCLE_WITH_DEFAULT = "Рендер круга c default";
const TEST_SNAPSHOT_CIRCLE_WITH_CHANGING = "Рендер круга c changing";
const TEST_SNAPSHOT_CIRCLE_WITH_MODIFIED = "Рендер круга c modified";

describe("Проверяем circle", () => {
  it(TEST_SNAPSHOT_CIRCLE_WITHOUT_LETTER, () => {
    const tree = renderer.create(<Circle letter="" />).toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITHOUT_LETTER);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_LETTERS, () => {
    const tree = renderer.create(<Circle letter="Сбукавами" />).toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_LETTERS);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_HEAD, () => {
    const tree = renderer.create(<Circle head="Check" />).toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_HEAD);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_HEAD_REACT, () => {
    const tree = renderer
      .create(
        <Circle
          head={<Button type="button" text="Test with react-element" />}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_HEAD_REACT);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_TAIL, () => {
    const tree = renderer.create(<Circle tail="test tail" />).toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_TAIL);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_TAIL_REACT, () => {
    const tree = renderer
      .create(
        <Circle
          tail={<Button type="button" text="Test with react-element" />}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_TAIL_REACT);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_INDEX, () => {
    const tree = renderer.create(<Circle index={34} />).toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_INDEX);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_SMALL, () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_SMALL);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_DEFAULT, () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_DEFAULT);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_CHANGING, () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_CHANGING);
  });

  it(TEST_SNAPSHOT_CIRCLE_WITH_MODIFIED, () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot(TEST_SNAPSHOT_CIRCLE_WITH_MODIFIED);
  });
});
