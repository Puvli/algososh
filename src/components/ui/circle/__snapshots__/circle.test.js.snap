// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяем circle Рендер круга c changing: Рендер круга c changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c default: Рендер круга c default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c head react: Рендер круга c head react 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      >
        Test with react-element
      </p>
    </button>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c head: Рендер круга c head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Check
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c index: Рендер круга c index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    34
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c modified: Рендер круга c modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c small: Рендер круга c small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга c tail react: Рендер круга c tail react 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      >
        Test with react-element
      </p>
    </button>
  </div>
</div>
`;

exports[`Проверяем circle Рендер круга c tail: Рендер круга c tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test tail
  </div>
</div>
`;

exports[`Проверяем circle Рендер круга c буквами: Рендер круга c буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Сбукавами
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем circle Рендер круга без буквы: Рендер круга без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c changing: Рендер круга c changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c default: Рендер круга c default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c head react: Рендер круга c head react 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      >
        Test with react-element
      </p>
    </button>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c head: Рендер круга c head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    Check
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c index: Рендер круга c index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    34
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Рендер круга c modified: Рендер круга c modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c small: Рендер круга c small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга c tail react: Рендер круга c tail react 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      >
        Test with react-element
      </p>
    </button>
  </div>
</div>
`;

exports[`Рендер круга c tail: Рендер круга c tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test tail
  </div>
</div>
`;

exports[`Рендер круга c буквами: Рендер круга c буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      Сбукавами
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Рендер круга без буквы: Рендер круга без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
